backend/app/services/answer_generator.py

```
@@ -368,12 +382,23 @@ def _format_filters_detailed(self, parsed_filter: Dict[str, Any]) -> str:
+        if ts and isinstance(ts, dict) and "$gte" in ts and "$lte" in ts:
             try:
-                start = datetime.fromisoformat(ts["$gte"])
-                end = datetime.fromisoformat(ts["$lte"])
+                start = datetime.fromisoformat(str(ts["$gte"]).replace('Z', '+00:00'))
+                end = datetime.fromisoformat(str(ts["$lte"]).replace('Z', '+00:00'))
+                start_fmt = start.strftime("%Y-%m-%d %I:%M:%S %p")
+                end_fmt = end.strftime("%Y-%m-%d %I:%M:%S %p")
                 duration_min = int((end - start).total_seconds() / 60)
-                parts.append(f"  Time Range: {ts['$gte']} to {ts['$lte']} ({duration_min} min)")
+                parts.append(f"  Time Range: {start_fmt} to {end_fmt} ({duration_min} min)")
+            except:
```

Ruff (0.15.1)

[error] Do not use bare `except`

(E722)